src/main/native/compression/IntelDeflater.cc

```diff
@@ -157,30 +166,44 @@ JNIEXPORT void JNICALL Java_com_intel_gkl_compression_IntelDeflater_resetNative
 
 JNIEXPORT void JNICALL Java_com_intel_gkl_compression_IntelDeflater_generateHuffman
 (JNIEnv * env, jobject obj) {
-
-      jbyteArray inputBuffer = (jbyteArray)env->GetObjectField(obj, FID_inputBuffer);
```

Comment:
To make the code readable I suggest we keep all the Get* vars together. Lets declare then check for error state it he order they were decarded and freeed as needed.

Reply:
This was intended for two reasons:
- I wanted to avoid unnecessarily accessing Java fields if an early exit condition exists (in this case it's `lz_stream` being `NULL`),
- this array is only being used in a single call to `GetPrimitiveArrayCritical` so I thought the code would be a little bit easier to follow if those two related calls happened immediately one after another.

One other thing to consider is that this operation could potentially fail (I think) if C and Java code drifted out of sync. An additional check could be made here (I think it's better to keep it out of this patch though) in which case the approach of first getting all the values and then checking for errors becomes invalid. If there's a pending JVM exception a lot of JNI calls become illegal (including all of the `Get*` methods).

Reply:
Agreed as discussed on Tuesday.